PalasoUIWindowsForms/ImageGallery/ImageUtilities.cs

```diff
@@ -24,58 +23,48 @@ public static Image GetThumbNail(string imagePath, int destinationWidth, int des
-				actualWidth = (int)(((float)bmp.Width / (float)bmp.Height) * actualHeight);
-
-			int horizontalOffset = (destinationWidth / 2) - (actualWidth / 2);
-			int verticalOffset = (destinationHeight / 2) - (actualHeight / 2);
-
-#if MONO
-//    this worked but didn't incorporate the offsets, so when it went back to the caller, it got displayed
-//            out of proportion.
-//            Image x = bmp.GetThumbnailImage(destinationWidth, destinationHeight, callbackOnAbort, System.IntPtr.Zero);
-//            return x;
+				actualWidth = (int)((bmp.Width / (float)bmp.Height) * actualHeight);
 
+			var horizontalOffset = (destinationWidth / 2) - (actualWidth / 2);
+			var verticalOffset = (destinationHeight / 2) - (actualHeight / 2);
 
-			Bitmap retBmp = new Bitmap(destinationWidth, destinationHeight);//, System.Drawing.Imaging.PixelFormat.Format64bppPArgb);
-			Graphics grp = Graphics.FromImage(retBmp);
-			//grp.PixelOffsetMode = PixelOffsetMode.None;
-		 //guessing that this is the problem?   grp.InterpolationMode = InterpolationMode.HighQualityBicubic;
-
-			grp.DrawImage(bmp, horizontalOffset, verticalOffset, actualWidth, actualHeight);
-
-//            Pen pn = new Pen(borderColor, 1); //Color.Wheat
-//
-//
-//            grp.DrawRectangle(pn, 0, 0, retBmp.Width - 1, retBmp.Height - 1);
-
-			return retBmp;
-#else
+			var retBmp = PlatformUtilities.Platform.IsWindows
+				? new Bitmap(destinationWidth, destinationHeight, System.Drawing.Imaging.PixelFormat.Format64bppPArgb)
+				: new Bitmap(destinationWidth, destinationHeight);
 
-			Bitmap retBmp = new Bitmap(destinationWidth, destinationHeight, System.Drawing.Imaging.PixelFormat.Format64bppPArgb);
-			Graphics grp = Graphics.FromImage(retBmp);
-			grp.PixelOffsetMode = PixelOffsetMode.None;
-			grp.InterpolationMode = InterpolationMode.HighQualityBicubic;
-
-			grp.DrawImage(bmp, horizontalOffset, verticalOffset, actualWidth, actualHeight);
-
-			Pen pn = new Pen(borderColor, 1); //Color.Wheat
-
-
-			grp.DrawRectangle(pn, 0, 0, retBmp.Width - 1, retBmp.Height - 1);
+			using (var grp = Graphics.FromImage(retBmp))
+			{
+				grp.PixelOffsetMode = PixelOffsetMode.None;
+				grp.InterpolationMode = InterpolationMode.HighQualityBicubic;
+
+				// fill with background color
+				using (var b = new SolidBrush(backgroundColor))
+				{
+					grp.FillRectangle(b, 0, 0, destinationWidth, destinationHeight);
+				}
+
+				// draw the image
+				grp.DrawImage(bmp, horizontalOffset, verticalOffset, actualWidth, actualHeight);
+
+				// draw border
+				using (var pn = new Pen(borderColor, 1))
+				{
+					grp.DrawRectangle(pn, 0, 0, retBmp.Width - 1, retBmp.Height - 1);
+				}
+			}
 
 			return retBmp;
-#endif
 		}
 
-		private static bool callbackOnAbort()
+		public static Image GetThumbNail(string imagePath, int destinationWidth, int destinationHeight, Color borderColor)
 		{
-			return false;
+			return GetThumbNail(imagePath, destinationWidth, destinationHeight, borderColor, Color.Black);
```

Comment:
Maybe, but I am trying to not change the previous behavior of the method. The reason I added the overload with background color is because the thumbnails were being created with a black background.